packages/kit/src/exports/vite/index.js

```diff
@@ -333,17 +261,152 @@ function kit({ svelte_config }) {
+				manifest_data.nodes.forEach((node) => {
+					for (const file of [node.component, node.universal, node.server]) {
+						if (file) {
+							const resolved = path.resolve(file);
+							const relative = decodeURIComponent(path.relative(kit.files.routes, resolved));
+
+							const name = relative.startsWith('..')
+								? posixify(path.join('entries/fallbacks', path.basename(file)))
+								: posixify(path.join('entries/pages', relative.replace(/\.js$/, '')));
+							input[name] = resolved;
+						}
+					}
+				});
+
+				// ...and every matcher
+				Object.entries(manifest_data.matchers).forEach(([key, file]) => {
+					const name = posixify(path.join('entries/matchers', key));
+					input[name] = path.resolve(file);
+				});
+			} else {
+				/** @type {Record<string, string>} */
+				input.start = `${runtime_directory}/client/start.js`;
+
+				manifest_data.nodes.forEach((node) => {
+					if (node.component) {
+						const resolved = path.resolve(node.component);
+						const relative = decodeURIComponent(
+							path.relative(svelte_config.kit.files.routes, resolved)
+						);
+
+						const name = relative.startsWith('..')
+							? path.basename(node.component)
+							: posixify(path.join('pages', relative));
+						input[`components/${name}`] = resolved;
+					}
+
+					if (node.universal) {
+						const resolved = path.resolve(node.universal);
+						const relative = decodeURIComponent(
+							path.relative(svelte_config.kit.files.routes, resolved)
+						);
+
+						const name = relative.startsWith('..')
+							? path.basename(node.universal)
+							: posixify(path.join('pages', relative));
+						input[`modules/${name}`] = resolved;
+					}
+				});
+			}
+
+			/** @type {import('vite').UserConfig} */
+			const new_config = {
+				base: ssr ? assets_base(kit) : './',
+				build: {
+					cssCodeSplit: true,
+					outDir: `${out}/${ssr ? 'server' : 'client'}`,
+					rollupOptions: {
+						input,
+						output: {
+							format: 'esm',
+							entryFileNames: ssr ? '[name].js' : `${prefix}/[name]-[hash].js`,
+							chunkFileNames: ssr ? 'chunks/[name].js' : `${prefix}/chunks/[name]-[hash].js`,
+							assetFileNames: `${prefix}/assets/[name]-[hash][extname]`,
+							hoistTransitiveImports: false
+						},
+						preserveEntrySignatures: 'strict'
+					},
+					target: ssr ? 'node16.14' : undefined,
+					// don't use the default name to avoid collisions with 'static/manifest.json'
+					manifest: 'vite-manifest.json'
+				},
+				define: {
+					__SVELTEKIT_ADAPTER_NAME__: JSON.stringify(kit.adapter?.name),
+					__SVELTEKIT_APP_VERSION_FILE__: JSON.stringify(`${kit.appDir}/version.json`),
+					__SVELTEKIT_APP_VERSION_POLL_INTERVAL__: JSON.stringify(kit.version.pollInterval),
+					__SVELTEKIT_DEV__: 'false',
+					__SVELTEKIT_EMBEDDED__: kit.embedded ? 'true' : 'false'
+				},
+				optimizeDeps: {
+					exclude: ['@sveltejs/kit']
+				},
+				publicDir: ssr ? false : kit.files.assets,
```

Comment:
this seems like something that belongs in the serve plugin and in fact it's being overridden there. does this one have any effect?

Reply:
errr.... not sure. but presumably the `ssr ? false : kit.files.assets` thing is for a reason. if it's not having an effect then perhaps something is broken? need to investigate

Reply:
Ah, this plugin only runs in build mode and the serve plugin only runs if not in build mode, so it won't override this. I'll resolve this comment